test/ShellWordsParseSpec.hs

```diff
@@ -6,7 +6,7 @@
 -- Case-for-case port from Python:
 --
 -- <https://github.com/mozillazg/python-shellwords/blob/master/tests/test_shellwords.py>
-module ShellWordsSpec
+module ShellWordsParseSpec
```

Comment:
This is a good signal that we should probably make `ShellWords.Parse` and `ShellWords.Quote` modules, with their own test-suites, then make `ShellWords` a re-export module.

Would you like to do that in this PR? Or I can do it after.

Reply:
Sure.

Reply:
Done, let me know if you'd prefer to have `ShellWords.Parse` and `ShellWords.Quote` listed as `other-modules` instead of exported.